### PR TITLE
Changed the nested loglikelihood function to a class method

### PR DESCRIPTION
**Pull request by shoepfl**
To parallelize dynesty, all objects have to be pickled.
Therefore, the loglikelihood was changed to a class method.
This enables pickling and parallelization via multiprocessing. An example script to use multiprocessing and dynesty with the pypesto wrapper will be uploaded to the according issue.

### CODE REVIEW

**Review by yannikschaelte — COMMENTED** [review]
looks good to me. @dilpath as this is your sampler, could you have a quick look?

File: pypesto/sample/dynesty.py [hunk 1]
```diff
@@ -118,7 +118,15 @@ def prior_transform(self, prior_sample: np.ndarray) -> np.ndarray:
             prior_sample * (self.problem.ub - self.problem.lb)
             + self.problem.lb
         )
-
+    def loglikelihood(self, x):
```
Comment (dilpath) [inline pypesto/sample/dynesty.py:121]:
```suggestion

    def loglikelihood(self, x):
```

**Review by dilpath — APPROVED** [review]
Thanks!

File: pypesto/sample/dynesty.py [hunk 2]
```diff
@@ -131,22 +140,14 @@ def initialize(
 
         self.problem = problem
```
Comment (dilpath) [inline pypesto/sample/dynesty.py:142]:
```suggestion
```

File: pypesto/sample/dynesty.py [hunk 3]
```diff
@@ -119,6 +119,15 @@ def prior_transform(self, prior_sample: np.ndarray) -> np.ndarray:
...
 
+    def loglikelihood(self, x):
+        """Log-probability density function."""
+        # check if parameter lies within bounds
+        if any(x < self.problem.lb) or any(x > self.problem.ub):
+            return -np.inf
+        # invert sign
+        # TODO this is possibly the posterior if priors are defined
+        return -1.0 * self.problem.objective(x)
+
```
Comment (dilpath) [inline pypesto/sample/dynesty.py:130]:
```suggestion

```

File: pypesto/sample/dynesty.py [hunk 4]
```diff
@@ -119,6 +119,14 @@ def prior_transform(self, prior_sample: np.ndarray) -> np.ndarray:
...
 
+    def loglikelihood(self, x):
+        """Log-probability density function."""
+        # check if parameter lies within bounds
+        if any(x < self.problem.lb) or any(x > self.problem.ub):
+            return -np.inf
+        # invert sign
+        # TODO this is possibly the posterior if priors are defined
+        return -1.0 * self.problem.objective(x)
     def initialize(
```
Comment (dilpath) [inline pypesto/sample/dynesty.py:130]:
```suggestion
        return -1.0 * self.problem.objective(x)

    def initialize(
```

**Pull request merged**